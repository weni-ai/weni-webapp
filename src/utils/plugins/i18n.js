import Vue from 'vue';
import VueI18n from 'vue-i18n';

import en from '../../locales/en.json';
import es from '../../locales/es.json';
import pt_br from '../../locales/pt_br.json';

Vue.use(VueI18n);

const languages = {
  'pt-br': pt_br,
  'pt-BR': pt_br,
  en: en,
  es: es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en',
  fallbackLocale: 'en',
  messages,
});

export default i18n;
