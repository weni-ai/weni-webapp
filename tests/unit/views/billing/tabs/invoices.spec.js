import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import Router from 'vue-router';
import Invoices from '@/views/billing/tabs/invoices.vue';
import i18n from '@/utils/plugins/i18n';
import { org } from '../../../../__mocks__';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Router);

const router = new Router();

jest.mock('@/services/SecurityService.js', () => {});
jest.mock('@/api/request.js', () => {});

describe('BillingInvoices.vue', () => {
  let wrapper;
  let store;
  let actions;

  beforeEach(() => {
    actions = {
      clearCurrentOrg: jest.fn(),
      clearCurrentProject: jest.fn(),
      createOrg: jest.fn(),
      changeAuthorization: jest.fn(),
      createProject: jest.fn(),
      setCurrentOrg: jest.fn(),
      setCurrentProject: jest.fn(),
      openModal: jest.fn(),
      setBillingOrgStep: jest.fn(),
      setBillingMembersStep: jest.fn(),
      setBillingProjectStep: jest.fn(),
      backBilling: jest.fn(),
    };

    store = new Vuex.Store({
      actions,
    });

    wrapper = shallowMount(Invoices, {
      localVue,
      i18n,
      store,
      router,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        RouterLink: RouterLinkStub,
        Indicator: true,
        UserManagement: true,
        Emoji: true,
        container: true,
        UnnnicButton: true,
        UnnnicSelect: true,
        UnnnicInput: true,
        UnnnicDatePicker:true,
        UnnnicTable: true,
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
