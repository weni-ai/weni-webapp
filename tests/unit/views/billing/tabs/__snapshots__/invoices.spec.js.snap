// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingInvoices.vue should be rendered properly 1`] = `
<div>
  <unnnictable-stub items="" class="invoices-table"></unnnictable-stub>
  <infinite-loading-stub loadingicon="true" hideerrorslot="true"></infinite-loading-stub>
</div>
`;
